docs/slurm-cluster/nvml.md

```diff
@@ -0,0 +1,58 @@
+GPU auto-detection with Slurm and NVML
+======================================
+
+By default, DeepOps auto-detects GPUs on each node using a [custom Ansible facts script](https://github.com/NVIDIA/deepops/blob/72fe3a187ceb36c76febb64c0bab484cbae6a451/roles/facts/files/gpus.fact),
+and uses this to generate Slurm configuration files such as [`slurm.conf`](https://slurm.schedmd.com/slurm.conf.html) and [`gres.conf`](https://slurm.schedmd.com/gres.conf.html).
+This mechanism detects GPUs using only `lspci`, allowing us to generate configuration even if a GPU driver is not present.
+
+The disadvantage to this method is that it generates static configuration files.
+This method cannot account for future changes in GPU hardware,
+or dynamic GPU configurations such as the use of [NVIDIA Multi-Instance GPU](https://www.nvidia.com/en-us/technologies/multi-instance-gpu/).
+
+Instead of generating the configuration based on `lspci` output, Slurm provides the option of [GPU auto-detection](https://slurm.schedmd.com/gres.conf.html#OPT_AutoDetect)
+using the [NVIDIA Management Library](https://developer.nvidia.com/nvidia-management-library-nvml) (NVML).
+This method enables Slurm to automatically detect the presence of NVIDIA GPUs, and set up local CPU and network affinity correctly.
+However, using this feature requires that the NVIDIA driver and CUDA be installed before building Slurm.
+
+
+## Turning on NVML auto-detection
+
+To enable NVML auto-detection in DeepOps, set the following variable in your DeepOps configuration:
+
+```
+slurm_autodetect_nvml: true
+```
+
+If building a new cluster, follow the [Slurm deployment guide](./README.md) to set up your new cluster.
+
+If enabling NVML on an existing cluster, run:
+
+```
+ansible-playbook -l slurm-cluster -e '{"slurm_force_rebuild": true}' playbooks/slurm-cluster/slurm.yml
+```
+
+
+## Configuring slurm.conf for heterogeneous configurations
```

Comment:
Should we add a link to the nvidia-mig.yml playbook or documentation around enabling/configuring MIG?

Similar to what I did in K8s https://github.com/NVIDIA/deepops/blob/master/docs/k8s-cluster/nvidia-mig.md

Reply:
That would be really helpful. I am not able to find the document related to enabling MIG and NVML.
Are there any prerequisite for NVML to work correctly?